wallet/keystores/encrypted_keystore.md

```diff
@@ -5,34 +5,33 @@ class EncryptedKeystore:
     // The constructor is not captured by the specs; it's up to the implementing library to define it.
 
     // Named constructor
-    static new_from_secret_key(secret_key: ISecretKey): EncryptedKeystore
+    static new_from_secret_key(secret_key: SecretKey, password: string, randomness: Optional[Randomnness]): EncryptedKeystore;
 
     // Named constructor
-    // Below, "wallet_provider" should implement "derive_secret_key_from_mnemonic()".
     // Advice: in the implementation all the parameters will be held as instance state (private fields).
-    static new_from_mnemonic(wallet_provider: IWalletProvider, mnemonic: Mnemonic): EncryptedKeystore
+    static new_from_mnemonic(mnemonic: Mnemonic, password: string, randomness: Optional[Randomnness]): EncryptedKeystore;
```

Comment:
> I think it sounds fine.
> 
> You can sketch these changes about the encrypted keystore in a different PR (not necessarily here in the specs, but in the implementation itself) - so that we follow the changes with higher care. CC @ccorcoveanu.

Yes - as a general rule, let's keep PRs focused